vengi-tools: fix build on darwin

booxter opened the pull request:
Also, move *.app bundles under /Applications/ directory.

Signed-off-by: Ihar Hrachyshka <ihar.hrachyshka@gmail.com>


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

booxter committed:
vengi-tools: fix build on darwin

Also, move *.app bundles under /Applications/ directory.

Signed-off-by: Ihar Hrachyshka <ihar.hrachyshka@gmail.com>

booxter commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 399609`

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>vengi-tools</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>vengi-tools</li>
  </ul>
</details>

booxter commented:
x86-64 build failure is an existing issue with darwin toolchain on 15.4:

```
[217/1028] Validate texture
FAILED: generated/shaders/render/TextureShader.h generated/shaders/render/TextureShader.cpp generated/shaders/render/TextureShaderConstants.h /tmp/nix-build-vengi-tools-0.0.35.d
rv-0/source/build/generated/shaders/render/TextureShader.h /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/generated/shaders/render/TextureShader.cpp /tmp/nix-build-vengi-t
ools-0.0.35.drv-0/source/build/generated/shaders/render/TextureShaderConstants.h 
cd /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/src/modules/render && /nix/store/x9390krmrb55ymmxp9xclnyhprbdim49-cmake-3.31.6/bin/cmake -E env APP_HOMEPATH=/tmp/nix-bui
ld-vengi-tools-0.0.35.drv-0/source/build/src/modules/render/ LSAN_OPTIONS=exitcode=0 /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/shadertool/vengi-shadertool -I /tmp/nix
-build-vengi-tools-0.0.35.drv-0/source/src/modules/render/shaders --shader /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/modules/render/shaders/texture --constantstemplate 
/tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/tools/shadertool/ShaderConstantsTemplate.h.in --headertemplate /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/tools/shader
tool/ShaderTemplate.h.in --sourcetemplate /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/tools/shadertool/ShaderTemplate.cpp.in --buffertemplate /tmp/nix-build-vengi-tools-0
.0.35.drv-0/source/src/tools/shadertool/UniformBufferTemplate.h.in --sourcedir /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/generated/shaders/render/
dyld[53381]: Symbol not found: __ZTINSt3__117bad_function_callE
  Referenced from: <68068DCF-9C83-3402-BC24-6446165606F4> /private/tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/shadertool/vengi-shadertool
  Expected in:     <F0AC1571-049A-3F10-BA96-5C6AD56E68C0> /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit
Subprocess aborted
[218/1028] Validate color
FAILED: generated/shaders/render/ColorShader.h generated/shaders/render/ColorShader.cpp generated/shaders/render/ColorShaderConstants.h /tmp/nix-build-vengi-tools-0.0.35.drv-0/s
ource/build/generated/shaders/render/ColorShader.h /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/generated/shaders/render/ColorShader.cpp /tmp/nix-build-vengi-tools-0.0.3
5.drv-0/source/build/generated/shaders/render/ColorShaderConstants.h 
cd /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/src/modules/render && /nix/store/x9390krmrb55ymmxp9xclnyhprbdim49-cmake-3.31.6/bin/cmake -E env APP_HOMEPATH=/tmp/nix-bui
ld-vengi-tools-0.0.35.drv-0/source/build/src/modules/render/ LSAN_OPTIONS=exitcode=0 /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/shadertool/vengi-shadertool -I /tmp/nix
-build-vengi-tools-0.0.35.drv-0/source/src/modules/render/shaders --shader /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/modules/render/shaders/color --constantstemplate /t
mp/nix-build-vengi-tools-0.0.35.drv-0/source/src/tools/shadertool/ShaderConstantsTemplate.h.in --headertemplate /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/tools/shaderto
ol/ShaderTemplate.h.in --sourcetemplate /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/tools/shadertool/ShaderTemplate.cpp.in --buffertemplate /tmp/nix-build-vengi-tools-0.0
.35.drv-0/source/src/tools/shadertool/UniformBufferTemplate.h.in --sourcedir /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/generated/shaders/render/
dyld[53165]: Symbol not found: __ZTINSt3__117bad_function_callE
  Referenced from: <68068DCF-9C83-3402-BC24-6446165606F4> /private/tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/shadertool/vengi-shadertool
  Expected in:     <F0AC1571-049A-3F10-BA96-5C6AD56E68C0> /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit
Subprocess aborted
[219/1028] Validate combine2
FAILED: generated/shaders/render/Combine2Shader.h generated/shaders/render/Combine2Shader.cpp generated/shaders/render/Combine2ShaderConstants.h /tmp/nix-build-vengi-tools-0.0.3
5.drv-0/source/build/generated/shaders/render/Combine2Shader.h /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/generated/shaders/render/Combine2Shader.cpp /tmp/nix-build-ve
ngi-tools-0.0.35.drv-0/source/build/generated/shaders/render/Combine2ShaderConstants.h 
cd /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/src/modules/render && /nix/store/x9390krmrb55ymmxp9xclnyhprbdim49-cmake-3.31.6/bin/cmake -E env APP_HOMEPATH=/tmp/nix-bui
ld-vengi-tools-0.0.35.drv-0/source/build/src/modules/render/ LSAN_OPTIONS=exitcode=0 /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/shadertool/vengi-shadertool -I /tmp/nix
-build-vengi-tools-0.0.35.drv-0/source/src/modules/render/shaders --shader /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/modules/render/shaders/combine2 --constantstemplate
 /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/tools/shadertool/ShaderConstantsTemplate.h.in --headertemplate /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/tools/shade
rtool/ShaderTemplate.h.in --sourcetemplate /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/tools/shadertool/ShaderTemplate.cpp.in --buffertemplate /tmp/nix-build-vengi-tools-
0.0.35.drv-0/source/src/tools/shadertool/UniformBufferTemplate.h.in --sourcedir /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/generated/shaders/render/
dyld[53240]: Symbol not found: __ZTINSt3__117bad_function_callE
  Referenced from: <68068DCF-9C83-3402-BC24-6446165606F4> /private/tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/shadertool/vengi-shadertool
  Expected in:     <F0AC1571-049A-3F10-BA96-5C6AD56E68C0> /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit
Subprocess aborted
[220/1028] Validate convolution
FAILED: generated/shaders/render/ConvolutionShader.h generated/shaders/render/ConvolutionShader.cpp generated/shaders/render/ConvolutionShaderConstants.h /tmp/nix-build-vengi-to
ols-0.0.35.drv-0/source/build/generated/shaders/render/ConvolutionShader.h /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/generated/shaders/render/ConvolutionShader.cpp /t
mp/nix-build-vengi-tools-0.0.35.drv-0/source/build/generated/shaders/render/ConvolutionShaderConstants.h 
cd /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/src/modules/render && /nix/store/x9390krmrb55ymmxp9xclnyhprbdim49-cmake-3.31.6/bin/cmake -E env APP_HOMEPATH=/tmp/nix-bui
ld-vengi-tools-0.0.35.drv-0/source/build/src/modules/render/ LSAN_OPTIONS=exitcode=0 /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/shadertool/vengi-shadertool -I /tmp/nix
-build-vengi-tools-0.0.35.drv-0/source/src/modules/render/shaders --shader /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/modules/render/shaders/convolution --constantstempl
ate /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/tools/shadertool/ShaderConstantsTemplate.h.in --headertemplate /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/tools/sh
adertool/ShaderTemplate.h.in --sourcetemplate /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/tools/shadertool/ShaderTemplate.cpp.in --buffertemplate /tmp/nix-build-vengi-too
ls-0.0.35.drv-0/source/src/tools/shadertool/UniformBufferTemplate.h.in --sourcedir /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/generated/shaders/render/
dyld[53309]: Symbol not found: __ZTINSt3__117bad_function_callE
  Referenced from: <68068DCF-9C83-3402-BC24-6446165606F4> /private/tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/shadertool/vengi-shadertool
  Expected in:     <F0AC1571-049A-3F10-BA96-5C6AD56E68C0> /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit
Subprocess aborted
[221/1028] Building CXX object src/modules/voxel/CMakeFiles/voxel.dir/MaterialColor.cpp.o
[222/1028] Building CXX object src/modules/voxel/CMakeFiles/voxel.dir/SurfaceExtractor.cpp.o
[223/1028] Building CXX object src/modules/palette/CMakeFiles/palette.dir/private/PixeloramaPalette.cpp.o
[224/1028] Building CXX object src/modules/voxel/CMakeFiles/voxel.dir/Face.cpp.o
[225/1028] Building CXX object src/modules/voxel/CMakeFiles/voxel.dir/private/MarchingCubesSurfaceExtractor.cpp.o
[226/1028] Building CXX object src/modules/voxel/CMakeFiles/voxel.dir/Mesh.cpp.o
[227/1028] Building CXX object src/modules/voxel/CMakeFiles/voxel.dir/private/CubicSurfaceExtractor.cpp.o
[228/1028] Building CXX object src/modules/voxel/CMakeFiles/voxel.dir/MeshState.cpp.o
ninja: build stopped: subcommand failed.

```

fgaz commented:
> x86-64 build failure is an existing issue with darwin toolchain on 15.4:

is this something that will be fixed in the toolchain, or should we mark this as broken for x86_64-darwin?

booxter commented:
@fgaz this is being worked on in https://github.com/NixOS/nixpkgs/pull/398727 I believe.

fgaz commented on pkgs/applications/graphics/vengi-tools/default.nix:
```diff
@@ -73,33 +79,35 @@ stdenv.mkDerivation (finalAttrs: {
+    if stdenv.hostPlatform.isDarwin then
+      ''
+        mkdir -p $out/Applications
+        for app in $out/*.app; do
+          mv "$app" $out/Applications/
+        done
```
```suggestion
        mv $out/*.app $out/Applications/
```

booxter commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 399609`

---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>vengi-tools</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>vengi-tools</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>vengi-tools</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>vengi-tools</li>
  </ul>
</details>

booxter committed:
vengi-tools: add /bin/vengi-voxconvert on darwin

It's a CLI tool and so should be available outside of an .app bundle.
This also fixes passthru.tests on darwin.

Signed-off-by: Ihar Hrachyshka <ihar.hrachyshka@gmail.com>

booxter commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 399609`

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>vengi-tools</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>vengi-tools</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>vengi-tools</li>
  </ul>
</details>

booxter commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 399609`

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>vengi-tools</li>
  </ul>
</details>

booxter commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 399609`

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>vengi-tools</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>vengi-tools</li>
  </ul>
</details>

booxter commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 399609`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>vengi-tools</li>
  </ul>
</details>

emilazy commented on pkgs/applications/graphics/vengi-tools/default.nix:
```diff
@@ -46,6 +43,15 @@ stdenv.mkDerivation (finalAttrs: {
   };
 
+  prePatch = lib.optionalString stdenv.hostPlatform.isDarwin ''
+    # Disable code signing on macOS
+    substituteInPlace cmake/macros.cmake --replace-fail "codesign" "true"
+    substituteInPlace cmake/system/apple.cmake --replace-fail "if(APPLE)" "if(false)"
```
FWIW, adding `sigtool` to the `nativeBuildInputs` may suffice.

emilazy commented on pkgs/applications/graphics/vengi-tools/default.nix:
```diff
@@ -46,6 +43,15 @@ stdenv.mkDerivation (finalAttrs: {
+    substituteInPlace cmake/macros.cmake --replace-fail "codesign" "true"
+    substituteInPlace cmake/system/apple.cmake --replace-fail "if(APPLE)" "if(false)"
+
+    # calls otool -L on /usr/lib/libSystem.B.dylib and fails because it doesn't exist
+    substituteInPlace cmake/applebundle.cmake --replace-fail 'fixup_bundle("''${TARGET_BUNDLE_DIR}" "" "")' ""
```
This is a CMake bug we can hopefully fix for 25.11, but other packages are already working around it, so this seems fine.

emilazy reviewed (approved):
Going to merge this to unblock Darwin deprecations for the 25.11 freeze deadline; hope you don’t mind, @fgaz!
